example/src/main/java/org/aerogear/mobile/example/ui/AuthDetailsFragment.java

```diff
@@ -1,26 +1,25 @@
 package org.aerogear.mobile.example.ui;
 
 import android.os.Bundle;
+import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
+import android.view.View;
 import android.widget.ArrayAdapter;
 import android.widget.ListView;
 import android.widget.TextView;
 
 import org.aerogear.mobile.auth.AuthService;
-import org.aerogear.mobile.auth.user.UserPrincipal;
+import org.aerogear.mobile.auth.user.UserPrincipalImpl;
 import org.aerogear.mobile.auth.user.UserRole;
 import org.aerogear.mobile.example.R;
 
-import java.util.ArrayList;
-import java.util.List;
-
 import butterknife.BindView;
 import butterknife.OnClick;
 
 public class AuthDetailsFragment extends BaseFragment {
     private static final String TAG = "AuthDetailsFragment";
 
-    public static final String PRINCIPAL = "USER-PRINCIPAL";
+    private UserPrincipalImpl currentUser;
```

Comment:
"Impl" classes are not meant to be public.
If we want user to be able to use them, we should remove the 'impl' suffix.
Better would be to deal only with the interface.

Reply:
addressed